### ISSUE TITLE
GUI doesn't start

### ISSUE THREAD

**Issue by kirienko** [kirienko/gourmet#4]
GUI cannot be started with existing `recipes.db`.

```bash
Traceback (most recent call last):
  File "bin/gourmet", line 17, in <module>
    GourmetRecipeManager.startGUI()
  File "./gourmet/GourmetRecipeManager.py", line 704, in startGUI
    r=RecGui(splash_label=splash.label)
  File "./gourmet/GourmetRecipeManager.py", line 924, in __init__
    editable=False)
  File "./gourmet/recindex.py", line 65, in __init__
    self.setup_widgets()
  File "./gourmet/recindex.py", line 106, in setup_widgets
    self.setup_rectree()
  File "./gourmet/recindex.py", line 222, in setup_rectree
    self.create_rmodel(self.rvw)
  File "./gourmet/recindex.py", line 217, in create_rmodel
    self.rmodel = RecipeModel(vw,self.rd,per_page=self.prefs.get('recipes_per_page',12))
  File "./gourmet/recindex.py", line 663, in __init__
    per_page=per_page)
  File "./gourmet/gtk_extras/pageable_store.py", line 323, in __init__
    per_page=per_page)
  File "./gourmet/gtk_extras/pageable_store.py", line 50, in __init__
    self.update_tree()
  File "./gourmet/gtk_extras/pageable_store.py", line 158, in update_tree
    self.append(row)
  File "/usr/lib/python3/dist-packages/gi/overrides/Gtk.py", line 971, in append
    return self._do_insert(-1, row)
  File "/usr/lib/python3/dist-packages/gi/overrides/Gtk.py", line 962, in _do_insert
    row, columns = self._convert_row(row)
  File "/usr/lib/python3/dist-packages/gi/overrides/Gtk.py", line 862, in _convert_row
    raise ValueError('row sequence has the incorrect number of elements')
ValueError: row sequence has the incorrect number of elements
```

**Comment by maweki** [comment]
There's some output before. This is the function in pageable_store that handles the exception:

```python
    def update_tree (self):
        """Update our tree based on current page, etc."""
        # clear the existing rows...
        for n in range(len(self)):
            self.remove(self.get_iter(0,))
        # and add the new ones...
        length = self._get_length_()
        start_at = self.page * self.per_page
        end_at = (self.page+1) * self.per_page
        if start_at > length: return # we're empty then...
        if end_at > length: end_at = length
        for row in self._get_slice_(int(start_at),int(end_at)):
            try:
                self.append(row)
            except (TypeError, ValueError) as e:
                print('columns          : ',self.columns)
                print('problem adding row ',row)
                raise
```

And I have this as example data:

```python
columns          :  ['rec', 'thumb', 'title', 'category', 'cuisine', 'rating', 'source', 'link', 'yields', 'yield_unit', 'preptime', 'cooktime']
problem adding row  [(1, 'Weihnachtlicher Nougat-Stollen - Rezept', '1 \n\n Den Nougat grob würfeln und in den Kühlschrank stellen. \n\n 2 \n200 Gramm Margarine cremig rühren, dann den Quark und den Zucker dazu geben u ... (1230 characters truncated) ... Eine Anmerkung zum Schluß, die Personenangabe ist Unsinn, der Stollen ist so groß, da wird ne ganz Kompanie satt!, aber man muß ja irgendwas angeben.', '', '', 0, None, '', 0, 0, None, 0.0, '', b'\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00\xff\xdb\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07\t\t\x08\n\x0c\x14\r\x0 ... (68881 characters truncated) ... \xff\x00\xaa\xab\xb7\xca\xd8\x1d=\r<\xf5?Z\x8d\xbe\xf8\xfci\xa2Hd\xc889\xa6\x8c\xe7\x8e\xdc\xd4\x93}\xf3P\x9e\xd5HD\x9b\xc7\xaf\xe9E2\x8a\x00\xff\xd9', b'\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00\xff\xdb\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07\t\t\x08\n\x0c\x14\r\x0 ... (4003 characters truncated) ... cd\x12I<g\x19\xc7LX\xf1\x16\xaf5\xba%\xbb]\xcd<\x8e\x9b\x11\xf1\xb70\x90p\x08\xe4gq\xecO\xe0z\x14Q{+\x8c\xf3\xf9fr\xe49%\xb3\x92OZ(\xa2\x91\'\xff\xd9', False, 'd2c2a4e4c6d65af216d815ca64841e84', '06db7a6c79ca31a09e75485773208295', 'https://www.kochbar.de/rezept/203952/Weihnachtlicher-Nougat-Stollen.html', 1543091386.78505), <GdkPixbuf.Pixbuf object at 0x7f2af112ef50 (GdkPixbuf at 0x56449466bb00)>, 'Weihnachtlicher Nougat-Stollen - Rezept', '', None, 0, None, 'https://www.kochbar.de/rezept/203952/Weihnachtlicher-Nougat-Stollen.html', None, None, 0, 0]
```

The column count seems to match up, at least the row. I haven't found where the model is defined.

**Comment by maweki** [comment]
The model is here: https://github.com/kirienko/gourmet/blob/master/gourmet/backends/db.py#L286-L316

But this also seems to match up with the record tuple.

**Comment by maweki** [comment]
Okay, so the issue is, that the types aren't handed off to the underlying listStore in https://github.com/kirienko/gourmet/blob/master/gourmet/gtk_extras/pageable_store.py#L28

that was caused by me because of the interface change I didn't know how to handle at the time:
https://github.com/kirienko/gourmet/commit/d93cd02f76f907ed441d018b237e128a530ebc59#diff-5cc813c06f6250ea3f65658bb93399c4R41

if `types` is correctly passed to the parent constructor, we should be good to go

**Comment by kirienko** [comment]
Fixed by #5
